bokeh/charts/_charts.py

```diff
@@ -144,17 +147,44 @@ def start_plot(self, xgrid, ygrid):
         if ygrid:
             self.make_grid(1, yaxis.ticker)
 
-        # Add tools
+        # Add tools if supposed to
         if self.tools:
+            # need to add tool to all underlying plots
             for plot in self._plots:
+                removed_tools = []
+                # only add tools if the underlying plot hasn't been customized
+                # by some user injection
                 if not plot.tools:
-                    if not self.categorical:
-                        pan = PanTool()
-                        wheelzoom = WheelZoomTool()
-                        reset = ResetTool()
-                        plot.add_tools(pan, wheelzoom, reset)
-                    previewsave = PreviewSaveTool()
-                    plot.add_tools(previewsave)
+                    tools = []
+                    # if tools have been specified as string let's reuse
+                    # plotting machinery to create the related tools
+                    if isinstance(self.tools, string_types):
+                        for tool in re.split(r"\s*,\s*", self.tools.strip()):
+                            # re.split will return empty strings; ignore them.
+                            if tool == "":
+                                continue
```

Comment:
>was there a reason why plotting was using self.tools.extend(tool_objs) while charts used plot.add_tools(*tool_objs). Any preference here? From an outsider look I prefer add_tools because it also provides some logic/validation...

`add_tools` was recently adopted (IIRC was written by @bryevdv) after some discussion about how to make easier for people to deal with the low level API... `self.tools.extend(tool_objs)` is an expression of this low level use... we should probably adopt one and unified the use...

>Shouldn't this just be a Plot method?

What do you mean? IIRC `add_tools` is a Plot method...